src/PositionManager.sol

```diff
@@ -227,7 +231,7 @@ contract PositionManager is
 
     // implementation of abstract function DeltaResolver._pay
     function _pay(Currency token, address payer, uint256 amount) internal override {
-        // TODO this should use Permit2
-        ERC20(Currency.unwrap(token)).safeTransferFrom(payer, address(poolManager), amount);
+        // TODO: Should we also support direct transfer?
```

Comment:
in v3 i think it was necessary because of exact input multihop case. what would be a use case for it now?

Reply:
also in UR we dont support sig transfer so maybe we dont need to here

Reply:
I think we do want to support transfer. Then UR can do 

removeLiquidity on v3 but send amounts directly to posm
and then call settle with transfer & not have to use permit2

if UR uses permit2 it has to approve permit2 which is not ideal i think

Reply:
got it! currently right now the payer is always the locker
```
        // the locker is the payer or receiver
        address caller = _getLocker();
        if (currencyDelta < 0) {
            _settle(currency, caller, uint256(-currencyDelta));
```
maybe there could be an extra param that specifies whether the payer is the address or the contract

Reply:
this was an issue in v3 periphery where the payer always had to be the msg.sender when adding liquidity
```abi.encode(MintCallbackData({poolKey: poolKey, payer: msg.sender}))```
https://github.com/Uniswap/v3-periphery/blob/main/contracts/base/LiquidityManagement.sol#L85

Reply:
yeah imo this will be 2 separate commands in posm but itll be interesting to see how they integrate into deltaresolver together... will you just use the same internal _pay function for both?
but either way that doesnt matter on this PR, just on the one where you do it haha